homeassistant/components/kostal_plenticore/select.py

```diff
@@ -20,31 +20,33 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-class SelectData(NamedTuple):
-    """Representation of a SelectData tuple."""
+@dataclass
+class PlenticoreRequiredKeysMixin:
+    """A class that describes required properties for plenticore select entities."""
 
     module_id: str
-    data_id: str
-    name: str
-    options: list
+    options: list[str]
     is_on: str
 
 
-# Defines all entities for select widgets.
-#
-# Each entry is defined with a tuple of these values:
-#  - module id (str)
-#  - process data id (str)
-#  - entity name suffix (str)
-#  - options
-#  - entity is enabled by default (bool)
+@dataclass
+class PlenticoreSelectEntityDescription(
+    SelectEntityDescription, PlenticoreRequiredKeysMixin
+):
```

Comment:
In this PR, I am only converting the `NamedTuple` to `SelectEntityDescription`.
The old `data_id` is mapped to `key`, and `name` is part of the standard `EntityDescription`.

Passing the `EntityDescription` to `PlenticoreDataSelect` will be done in a follow-up PR.